File: codes/script_threads/run_call.r
sink(paste(opt$output, ".call", sep=""))
cat('Call:', '\n')
#cat('Rscript stars/run.r -F', opt$input,           #modified M.Zigo 20200116
cat('Rscript script_threads/run.r -F', opt$input, 
                        '-A', opt$width,
                        '-B', opt$height,
                        '-C', opt$angle,
                        '-N', opt$`noise-dim`,
                        '-L', opt$`local-noise`,
                        '-D', opt$delta,
                        '-X', opt$`start-iter`,
                        '-M', opt$`max-iter`,
                        '-I', opt$`min-iter`,
                        '-S', opt$`snr-lim`,
                        '-Z', opt$color,
                        '-E', opt$model,
                        '-O', opt$output,
                        '-Y', opt$`cent-pix-perc`,
                        '-G', opt$`init-noise-removal`,
                        '-H', opt$`fine-iter`,
                        '-K', opt$method,
                        '-P', opt$parallel,
                        '-V', opt$verbose,
                        '\n\n'
)
sink()

File: codes_python/utils/run_call.py
def save_call(args):
    data = 'python3 ./main.py '
    data += '-F ' + str(args.input) + ' '
    data += '-A ' + str(args.width) + ' '
    data += '-B ' + str(args.height) + ' '
    data += '-C ' + str(args.angle) + ' '
    data += '-N ' + str(args.noise_dim) + ' '
    data += '-L ' + str(args.local_noise) + ' '
    data += '-D ' + str(args.delta) + ' '
    data += '-X ' + str(args.start_iter) + ' '
    data += '-M ' + str(args.max_iter) + ' '
    data += '-I ' + str(args.min_iter) + ' '
    data += '-S ' + str(args.snr_lim) + ' '
    data += '-Z ' + str(args.color) + ' '
    if args.model != '':
        data += '-E ' + str(args.model) + ' '
    data += '-O ' + str(args.output) + ' '
    data += '-Y ' + str(args.cent_pix_perc) + ' '
    data += '-G ' + str(args.init_noise_removal) + ' '
    data += '-H ' + str(args.fine_iter) + ' '
    data += '-K ' + str(args.method) + ' '
    data += '-P ' + str(args.parallel) + ' '
    data += '-V ' + str(args.verbose) + '\n'

    with open(args.output+'.call', 'a') as f:
        print(data, file=f)